weblate/machinery/microsoft.py

```diff
@@ -18,19 +18,27 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
+import json as _js
 from datetime import timedelta
 
 from django.conf import settings
 from django.utils import timezone
 
 from weblate.machinery.base import MachineTranslation, MissingConfiguration
 
-COGNITIVE_BASE_URL = 'https://api.cognitive.microsoft.com/sts/v1.0'
+try:
+    if settings.MT_MICROSOFT:
```

Comment:
Please avoid using settings here. This is executed at import time, when settings might not be yet initialized or might have wrong value. Move this logic to `__ini__` method and construct the url in the class.